avocado/core/nrunner.py

```diff
@@ -424,23 +434,22 @@ def _run_unittest(uri, queue):
         output = {'status': 'finished',
                   'result': result,
                   'output': stream.read(),
-                  'time_end': time_end}
+                  'time': time_end}
```

Comment:
This will be overwritten by https://github.com/avocado-framework/avocado/pull/3723/commits/12488c9a5508e9a6e50a4903c6cc157e7cab2c3f#diff-5a2f92aedf41bef8d89fc436e1b3c84cR296, won't it?

We have to decide if we'll let the subclasses have the final saying in the time (like in this example), and I think they should, as this would be more accurate.

Reply:
@clebergnu, to avoid having some runners without the information or with a different key ('time' vs 'timestamp' vs 'time_start' vs 'time_end')  I thought that is worthed to create a convention here: *Every message will have the 'time' key.*

I understand that if we let this in the subclass we will have a "more accurate" time, but I did a few runs comparing both, and this is the difference that we are talking about:

```
direct: 1587139446.4457245
indirect: 1587139446.4457424
```

Unless that I'm missing something, I think that putting this inside a common method will simplify the code and ensure that all messages will have the time (in case some runner forgot about it). Also with a common method is easy to add default values in the future for specific cases if needed, only one place change.

But I confess that I'm not sure if this small difference in time will impact here. What do you think?

Reply:
> @clebergnu, to avoid having some runners without the information or with a different key ('time' vs 'timestamp' vs 'time_start' vs 'time_end') I thought that is worthed to create a convention here: _Every message will have the 'time' key._
> 

This is fine, and it's a good idea. Still, it doesn't conflict with some runner subclasses wanting to provide the time themselves, right? IMO, It's just a matter of the base class method not overwriting something if the subclass set it.

> I understand that if we let this in the subclass we will have a "more accurate" time, but I did a few runs comparing both, and this is the difference that we are talking about:
> 
> ```
> direct: 1587139446.4457245
> indirect: 1587139446.4457424
> ```
> 

This is a very small difference, on such a small test and on an underloaded system. It may be different, and relevant, on different test environment.

> Unless that I'm missing something, I think that putting this inside a common method will simplify the code and ensure that all messages will have the time (in case some runner forgot about it). Also with a common method is easy to add default values in the future for specific cases if needed, only one place change.
> 
> But I confess that I'm not sure if this small difference in time will impact here. What do you think?

I just think a subclass should have the final say about its status messages and the `time` is just one example.  I do agree that is not very important right now because we don't have tests running on loaded system reporting, say, +/-30% different execution times, so it's really about the hypothetical issue.

Other than that, do you think that a runner passing `time` to `prepare_status` and finding that it got changed is correct, implementation wise?

Reply:
> This is fine, and it's a good idea. Still, it doesn't conflict with some runner subclasses wanting to provide the time themselves, right? IMO, It's just a matter of the base class method not overwriting something if the subclass set it.

Yes, I agree. And I can add an IF to base class to check if there is no 'time'. 

One problem that I can see with this approach is the lack of a "convention". "base class" can try to see if `time` is set but the runner can use a different name like `my_custom_timer`. So everything is about a convention and in order to avoid that I was trying to centralize the things here.

> Other than that, do you think that a runner passing `time` to `prepare_status` and finding that it got changed is correct, implementation wise?

No, it isn't correct. And that was, not the idea. The idea is to not pass `time` to `prepare_status`, but `prepare_status` create the time for the runner. The runner will say: give-me a status_message. Like "generate an event".

Reply:
Right, so I think you can (instead of an `if`) just start with the status the subclass is giving and `dict.update()` it with the information that is set by convention/default on `prepare_status()`.

Reply:
@clebergnu IIUC with your suggestion will bring back the problem that you point me out: the base class will override the time provided by the runner.

```python
default = {'time': 2} # base class
additional_info = {'time': 1} # subclass (runner)
addititonal_info.update(default)
print(additional_info['time'])
2
```

Inverting this logic (start with default and update with that is provided by the runner) is the current implementation.

Reply:
> @clebergnu IIUC with your suggestion will bring back the problem that you point me out: the base class will override the time provided by the runner.
> 
> ```python
> default = {'time': 2} # base class
> additional_info = {'time': 1} # subclass (runner)
> addititonal_info.update(default)
> print(additional_info['time'])
> 2
> ```
> 
> Inverting this logic (start with default and update with that is provided by the runner) is the current implementation.

I was just testing your attention! :smile: 

Now, seriously, I meant that but clearly got it wrong when trying to explain it.